test/number-field.html

```diff
@@ -251,6 +251,26 @@
 
           expect(numberField.value).to.be.equal(numberField.min);
         });
+
+        it('should decrease value to min value when step is over the min value', () => {
+          numberField.value = 2;
+          numberField.min = 1;
+          numberField.step = 10;
+
+          decreaseButton.click();
+
+          expect(numberField.value).to.be.equal('1');
+        });
+
+        it('should increase value to max value when step is over the max value', () => {
+          numberField.value = 1;
+          numberField.max = 2;
+          numberField.step = 10;
+
+          increaseButton.click();
+
+          expect(numberField.value).to.be.equal('2');
```

Comment:
I don't think this is the right assumption. If step is 10, then 2 would be an invalid value. See how native `<input type="number">` works in this situation and align the behaviour.